src/main/java/at/ac/tuwien/kr/alpha/grounder/NaiveGrounder.java

```diff
@@ -140,6 +164,7 @@ private void initializeFactsAndRules(Program program) {
 			// Record the rule for later use
 			NonGroundRule nonGroundRule = NonGroundRule.constructNonGroundRule(rule);
 			knownNonGroundRules.put(nonGroundRule.getRuleId(), nonGroundRule);
+			LOGGER.debug("NonGroundRule #" + nonGroundRule.getRuleId() + ": " + nonGroundRule);
```

Comment:
Please use placeholders `{}` in the debug string (and var-args) to avoid unnecessary calls of `toString()`.

Reply:
Done, thanks for the hint!